src/de/jost_net/JVerein/Queries/MitgliedQuery.java

```diff
@@ -438,26 +393,133 @@ else if (sort.equals("Geburtstagsliste"))
+      {
+        String s = stt.nextToken();
+        Long id = Long.valueOf(s.substring(0,s.length()-1));
+        suchIds.add(id);
+        suchauswahl.put(id, s.substring(s.length()-1));
+      }    
+      
+      // Eigenschaften lesen
+      String sql = "SELECT eigenschaften.* from eigenschaften ";
+      List<Long[]> mitgliedeigenschaften = (List<Long[]>) service.execute(sql,
+          new Object[] { }, new ResultSetExtractor()
+      {
+        @Override
+        public Object extract(ResultSet rs) throws RemoteException, SQLException
+        {
+          List<Long[]> list = new ArrayList<>();
+          while (rs.next())
+          {
+            list.add(new Long[] {rs.getLong(2), rs.getLong(3)}); // Mitglied.Id, Eigenschaft.Id
+          }
+          return list;
+        }
+      });
+      
+      ArrayList<Long> mitgliederIdsNeu = new ArrayList<>();
+      for (Long mitglied: mitgliederIds)
+      {
+        ArrayList<Long> mitgliedeigenschaftenIds = new ArrayList<>();
+        for (Long[] value: mitgliedeigenschaften)
+        {
+          if (value[0].equals(mitglied))
+            mitgliedeigenschaftenIds.add(value[1]);
+        }
+
+        boolean ok = false;
+        for (Long suchId : suchIds)
+        {
+          if (control.getEigenschaftenVerknuepfung().equals("und"))
+          {
+            ok = true;
+            if(suchauswahl.get(suchId).equals(EigenschaftenNode2.PLUS))
+            {
+              if (!mitgliedeigenschaftenIds.contains(suchId))
+              {
+                ok = false;
+                break;
+              }
+            }
+            else  // EigenschaftenNode2.MINUS
+            {
+              if (mitgliedeigenschaftenIds.contains(suchId))
+              {
+                ok = false;
+                break;
+              }
+            }
+          }
+          else    // Oder
+          {
+            if(suchauswahl.get(suchId).equals(EigenschaftenNode2.PLUS))
+            {
+              if (mitgliedeigenschaftenIds.contains(suchId))
+              {
+                ok = true;
+                break;
+              }
+            }
+            else  // EigenschaftenNode2.MINUS
+            {
+              if (!mitgliedeigenschaftenIds.contains(suchId))
+              {
+                ok = true;
+                break;
+              }
+            }
+          }
+        }
+        if (ok)
+          mitgliederIdsNeu.add(mitglied);
+      }
+      return getMitglieder(mitgliederIdsNeu);
+    }
+    else
+    {
+      return getMitglieder(mitgliederIds);
+    }
+  }
+  
+  private ArrayList<Mitglied> getMitglieder(ArrayList<Long> ids) 
+      throws RemoteException
+  {
+    
+    DBIterator<Mitglied> list = Einstellungen.getDBService().createList(Mitglied.class);
+    list.addFilter("id in (" + StringUtils.join(ids, ",") + ")");
+    @SuppressWarnings("unchecked")
+    ArrayList<Mitglied> mitglieder = list != null ? (ArrayList<Mitglied>) PseudoIterator.asList(list) : null;;
```

Comment:
`Einstellungen.getDBService().createList`
gibt nie null zurück, höchstens einen leeren Iterator, von daher könntest du hier einfach
`return (ArrayList<Mitglied>) PseudoIterator.asList(list);`
machen

Reply:
Es gab den PR #162 von mir. Dort gab es dann eine null Pointer Exception da der iterator anscheinend doch null sein konnte. Ich habe dann das in #197 korrigiert. Seitdem benutze ich diese Zuweisung so.

Reply:
ok, aus dem Code von DBIterator habe ich das nicht gesehen. Dann bleibt es so.